iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/ResourceDeclarationConverter.java

```diff
@@ -65,28 +67,47 @@ public ResourceDeclaration convertToResourceDeclaration(MappingTree tree) {
     StringLiteral type = toStringLiteralOrException(tree, "type");
     StringLiteral version = toStringLiteralOrException(tree, "apiVersion");
     Identifier name = toIdentifierOrException(tree, "name");
+    List<Property> resourceProperties = toResourceProperties(tree);
     List<Property> otherProperties = PropertyUtils.get(tree, "properties")
       .map(PropertyTree::value)
       .map(this::toProperties)
       .orElse(Collections.emptyList());
 
     return PropertyUtils.get(tree, "resources")
-      .map(childResources -> toResourceDeclarationWithChildren(type, version, name, otherProperties, childResources))
-      .orElseGet(() -> toResourceDeclaration(type, version, name, otherProperties));
+      .map(childResources -> toResourceDeclarationWithChildren(type, version, name, otherProperties, resourceProperties, childResources))
+      .orElseGet(() -> toResourceDeclaration(type, version, name, otherProperties, resourceProperties));
   }
 
-  private static ResourceDeclaration toResourceDeclaration(StringLiteral type, StringLiteral version, Identifier name, List<Property> otherProperties) {
-    return new ResourceDeclarationImpl(name, version, type, otherProperties, Collections.emptyList());
+  private List<Property> toResourceProperties(MappingTree tree) {
+    return tree.elements().stream()
+      .map(tupleTree -> {
+        Identifier key = toIdentifier(tupleTree.key());
+        Expression value = toExpression(tupleTree.value());
+        return new PropertyImpl(key, value);
+      })
+      .collect(Collectors.toList());
+  }
```

Comment:
Will this `resourceProperties` contain also `properties`?
If so, should we think about deprecating our old `properties`, and in the near future replace it with `resourceProperties`?

Reply:
Yes, that's why I wanted to propose to keep your code as it is and just add a `@Deprecated` tag on the method properties(), as now with `resourceProperties()` we have two ways to access the same data, and the one you are proposing is more complete.
I don't want to change everything now, just potentially encourage writers of next rules to use this `resourceProperties()` instead of `properties()`.